src/core/server/ui_settings/settings/date_formats.ts

```diff
@@ -0,0 +1,152 @@
+  ];
+
+  return {
+    dateFormat: {
+      name: i18n.translate('kbn.advancedSettings.dateFormatTitle', {
+        defaultMessage: 'Date format',
+      }),
+      value: 'MMM D, YYYY @ HH:mm:ss.SSS',
+      description: i18n.translate('kbn.advancedSettings.dateFormatText', {
+        defaultMessage: 'When displaying a pretty formatted date, use this {formatLink}',
+        description:
+          'Part of composite text: kbn.advancedSettings.dateFormatText + ' +
+          'kbn.advancedSettings.dateFormat.optionsLinkText',
+        values: {
+          formatLink:
+            '<a href="https://momentjs.com/docs/#/displaying/format/" target="_blank" rel="noopener noreferrer">' +
+            i18n.translate('kbn.advancedSettings.dateFormat.optionsLinkText', {
+              defaultMessage: 'format',
+            }) +
+            '</a>',
+        },
+      }),
+      schema: schema.string(),
+    },
+    'dateFormat:tz': {
+      name: i18n.translate('kbn.advancedSettings.dateFormat.timezoneTitle', {
+        defaultMessage: 'Timezone for date formatting',
+      }),
+      value: 'Browser',
+      description: i18n.translate('kbn.advancedSettings.dateFormat.timezoneText', {
+        defaultMessage:
+          'Which timezone should be used. {defaultOption} will use the timezone detected by your browser.',
+        values: {
+          defaultOption: '"Browser"',
+        },
+      }),
+      type: 'select',
+      options: timezones,
+      requiresPageReload: true,
+      schema: schema.string({
+        validate: (value) => {
+          if (!timezones.includes(value)) {
+            return `Invalid timezone: ${value}`;
+          }
+        },
+      }),
+    },
+    'dateFormat:scaled': {
+      name: i18n.translate('kbn.advancedSettings.dateFormat.scaledTitle', {
+        defaultMessage: 'Scaled date format',
+      }),
+      type: 'json',
+      value: `[
+  ["", "HH:mm:ss.SSS"],
+  ["PT1S", "HH:mm:ss"],
+  ["PT1M", "HH:mm"],
+  ["PT1H", "YYYY-MM-DD HH:mm"],
+  ["P1DT", "YYYY-MM-DD"],
+  ["P1YT", "YYYY"]
+]`,
+      description: i18n.translate('kbn.advancedSettings.dateFormat.scaledText', {
+        defaultMessage:
+          'Values that define the format used in situations where time-based ' +
+          'data is rendered in order, and formatted timestamps should adapt to the ' +
+          'interval between measurements. Keys are {intervalsLink}.',
+        description:
+          'Part of composite text: kbn.advancedSettings.dateFormat.scaledText + ' +
+          'kbn.advancedSettings.dateFormat.scaled.intervalsLinkText',
+        values: {
+          intervalsLink:
+            '<a href="http://en.wikipedia.org/wiki/ISO_8601#Time_intervals" target="_blank" rel="noopener noreferrer">' +
+            i18n.translate('kbn.advancedSettings.dateFormat.scaled.intervalsLinkText', {
+              defaultMessage: 'ISO8601 intervals',
+            }) +
+            '</a>',
+        },
+      }),
+      schema: schema.string(),
```

Comment:
Same here, no obvious way to validate the format. Parsing the json value to check if it's an array of array[2] felt overkill (and is not enough)